### DISCUSSION TITLE
adding prometheus-client support on fastapi

### DISCUSSION THREAD

**Discussion by blnprasad** [fastapi/fastapi#4724] [Questions] [answered]
Hi,
   I'm planning to integrate prometheus-client to my fastapi application and serve metrics under path "/metrics". Their documentation 
says we should use their asgi_app some thing similar to below . can you please give me  equalant  of below for fastapi

from prometheus_client import make_wsgi_app

******* Create my app
app = Flask(__name__)

******* Add prometheus wsgi middleware to route /metrics requests
app.wsgi_app = DispatcherMiddleware(app.wsgi_app, {
    '/metrics': make_wsgi_app()
})

Thanks,
Prasad

**Comment by YuriiMotov** [comment, marked as answer]
See https://github.com/trallnag/prometheus-fastapi-instrumentator

```python
from fastapi import FastAPI
from prometheus_fastapi_instrumentator import Instrumentator

app = FastAPI()

# Instrument the app
Instrumentator().instrument(app).expose(app, endpoint="/metrics")
```
